test/functorch/test_aotdispatch.py

```diff
@@ -5989,6 +5990,222 @@ def forward(self, x):
+
+            def forward(self, x):
+                r = self.conv(x)
+                return r
+
+        m = M()
+        m.train()
+
+        def inps_fn(x):
+            return (
+                TwoTensor(
+                    TwoTensor(x.clone(), x.clone()), TwoTensor(x.clone(), x.clone())
+                ),
+            )
+
+        benchmark_inps = [
+            inps_fn(
+                torch.randn(2, 3, 5, 5, requires_grad=True).to(
+                    memory_format=torch.channels_last
+                )
+            )
+            for _ in range(num_iters)
+        ]
+
+        for i, inps in enumerate(benchmark_inps):
+            outs = torch.compile(m, backend="aot_eager", fullgraph=True)(*inps)
+            outs[0].sum().backward()
+
+        def _assert_len():
+            assert len(fwd_compiled_module_durations_ns) == num_iters
+            assert len(fwd_overheads_ns) == num_iters
+            assert len(fwd_total_durations_ns) == num_iters
+            assert len(bwd_compiled_module_durations_ns) == num_iters
+            assert len(bwd_overheads_ns) == num_iters
+            assert len(bwd_total_durations_ns) == num_iters
+            assert len(fwd_runtime_wrapper_durations_ns) == num_iters
+
+        _assert_len()
+
+        def _print_stat(s: str):
+            fwd_runtime_wrapper_avg_ns = (
+                sum(fwd_runtime_wrapper_durations_ns) / num_iters
+            )
+            fwd_compiled_module_duration_avg_ns = (
+                sum(fwd_compiled_module_durations_ns) / num_iters
+            )
+            fwd_aotd_overhead_avg_ns = sum(fwd_overheads_ns) / num_iters
+            fwd_runtime_wrapper_duration_avg_ns = (
+                sum(fwd_runtime_wrapper_durations_ns) / num_iters
+            )
+            print(
+                f"{s} FW_aotd_overhead_avg:{fwd_aotd_overhead_avg_ns:.3f}ns "
+                f"{100 * fwd_aotd_overhead_avg_ns / fwd_compiled_module_duration_avg_ns:.3f}% "
+                f"of fwd_compiled_module_duration_avg_ns:{fwd_compiled_module_duration_avg_ns:.3f}ns "
+                f"fwd_runtime_wrapper_duration_avg_ns:{fwd_runtime_wrapper_duration_avg_ns:.3f}ns"
+            )
+
+            bwd_compiled_module_duration_avg_ns = (
+                sum(bwd_compiled_module_durations_ns) / num_iters
+            )
+            bwd_aotd_overhead_avg_ns = sum(bwd_overheads_ns) / num_iters
+            bwd_total_duration_avg_ns = sum(bwd_total_durations_ns) / num_iters
+            print(
+                f"{s} BW_aotd_overhead_avg:{bwd_aotd_overhead_avg_ns:.3f}ns "
+                f"{100 * bwd_aotd_overhead_avg_ns / bwd_compiled_module_duration_avg_ns:.3f}% "
+                f"of bwd_compiled_module_duration_avg:{bwd_compiled_module_duration_avg_ns:.3f}ns "
+                f"bwd_total_duration_avg:{bwd_total_duration_avg_ns:.3f}ns"
+            )
+
+        # Uncomment for adhoc aotd overhead perf experimentation
+        _print_stat("   SUBCLASS")
```

Comment:
Is this meant to be commented out before you check it in (So we don't print during CI runs by default)?

Also, I think you mentioned in our 1:1 - but were you thinking of adding this into a benchmark somewhere that we can track over time for regressions? (e.g. we have some benchmarks here, although they are mainly compile-time related https://github.com/pytorch/pytorch/tree/main/benchmarks/dynamo/pr_time_benchmarks/benchmarks)